### PR TITLE
Fixed exports so internal classes are available for static typing

### PR DESCRIPTION
**Pull request by maniacalmaniac**

### CODE REVIEW

**Comment by mrand01** [comment]
PR covers 2 different projects - apologies

**Comment by vvakame** [comment]
*moment/moment-node.d.ts*

to authors(@Lakerfield @kingdango @horiuchi @DickvdBrink @adidahiya @EnableSoftware). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

------

*xmlbuilder/xmlbuilder.d.ts*

to author(@wallymathieu). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

File: xmlbuilder/xmlbuilder-tests.ts [hunk 1]
```diff
@@ -14,31 +17,31 @@ xml('test17', { headless: true }).ins({ 'pi': 'mypi', 'pi2': 'mypi2', 'pi3': nul
 xml('test17', { headless: true }).ins(['pi', 'pi2']).end();
 
 xml('test18', { headless: true })
-    .ins('renderCache.subset', '"Verdana" 0 0 ISO-8859-1 4 268 67 "#(),-./')
-    .ins('pitarget', () => 'pivalue')
-    .end();
+  .ins('renderCache.subset', '"Verdana" 0 0 ISO-8859-1 4 268 67 "#(),-./')
+  .ins('pitarget', () => 'pivalue')
+  .end();
```
Comment (adidahiya) [inline xmlbuilder/xmlbuilder-tests.ts]: these whitespace changes are unnecessary. please revert to keep a cleaner history

File: xmlbuilder/xmlbuilder.d.ts [hunk 2]
```diff
@@ -4,86 +4,83 @@
...
-        entity(name: string, value: any): XMLDocType;
-        pEntity(name: string, value: any): XMLDocType;
-        notation(name: string, value: any): XMLDocType;
-        cdata(value: string): XMLDocType;
-        comment(value: string): XMLDocType;
-        instruction(target: string, value: any): XMLDocType;
-        root(): XMLDocType;
-        document(): any;
-        toString(options?: Object, level?: Number): string;
+  export class XMLDocType {
```
Comment (adidahiya) [inline xmlbuilder/xmlbuilder.d.ts]: same here, there is no need for all these whitespace changes; it results in a bad diff / history tracking.

File: moment/moment-node.d.ts [hunk 3]
```diff
@@ -374,7 +374,22 @@ declare module moment {
...
+        default(date: string, format?: string, language?: string, strict?: boolean): Moment;
+        default(date: string, formats: string[], strict?: boolean): Moment;
+        default(date: string, formats: string[], language?: string, strict?: boolean): Moment;
+        default(date: string, specialFormat: () => void, strict?: boolean): Moment;
+        default(date: string, specialFormat: () => void, language?: string, strict?: boolean): Moment;
+        default(date: string, formatsIncludingSpecial: any[], strict?: boolean): Moment;
+        default(date: string, formatsIncludingSpecial: any[], language?: string, strict?: boolean): Moment;
+        default(date: Date): Moment;
+        default(date: Moment): Moment;
+        default(date: Object): Moment;
```
Comment (adidahiya) [inline moment/moment-node.d.ts]: where does this method come from? I don't see it in the moment docs. also there are no tests for this

**Comment by rob3c** [comment]
Hi,
I think the biggest argument for this important change - even if it's breaking - is being able to preserve strong typing while avoiding single large functions that create xml documents. We need the inner classes exposed so we can break up xml doc and node creation into smaller, targeted functions by being able to type their return values with XMLDocType or XMLElementOrXMLNode.

In the meantime, I'm using a custom corrected version similar to what has already been submitted as a pull request. I see this as yet another reason for TSD to allow pulling from other registries besides this github repo, since not everyone may agree on the best way to write type definitions.

**Comment by RyanCavanaugh** [comment]
Hello and thank you for your contribution!

Due to an excessively long queue of pull requests that have become stale, we are "declaring bankruptcy" and closing all PRs opened before May 1, 2016. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

Going forward, we are committing to review or merge all PRs on a regular basis so this bankruptcy will not occur again. We apologize for the incovenience and hope you will continue to contribute to DefinitelyTyped in the future.

Thanks
The TypeScript and DefinitelyTyped teams

**Pull request closed without merging**